小米 5X 7.1.2 调用 AndPermission.with(this) .install() 出现解析包时出现错误（应用内更新）

07-03 16:20:03.295 1523-2801/? W/ActivityManager: Permission Denial: opening provider com.yanzhenjie.permission.FileProvider from ProcessRecord{90cc145 27283:com.miui.packageinstaller/u0a22} (pid=27283, uid=10022) that is not exported from uid 10280
07-03 16:20:03.296 27283-4180/? W/PackageInstaller: Error staging apk from content URI
                                                    java.lang.SecurityException: Permission Denial: opening provider com.yanzhenjie.permission.FileProvider from ProcessRecord{90cc145 27283:com.miui.packageinstaller/u0a22} (pid=27283, uid=10022) that is not exported from uid 10280
                                                        at android.os.Parcel.readException(Parcel.java:1684)
                                                        at android.os.Parcel.readException(Parcel.java:1637)
                                                        at android.app.ActivityManagerProxy.getContentProvider(ActivityManagerNative.java:4278)
                                                        at android.app.ActivityThread.acquireProvider(ActivityThread.java:5665)
                                                        at android.app.ContextImpl$ApplicationContentResolver.acquireUnstableProvider(ContextImpl.java:2278)
                                                        at android.content.ContentResolver.acquireUnstableProvider(ContentResolver.java:1533)
                                                        at android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:1146)
                                                        at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:999)
                                                        at android.content.ContentResolver.openInputStream(ContentResolver.java:719)
                                                        at com.android.packageinstaller.PackageInstallerActivity$StagingAsyncTask.doInBackground(PackageInstallerActivity.java:1079)
                                                        at com.android.packageinstaller.PackageInstallerActivity$StagingAsyncTask.doInBackground(PackageInstallerActivity.java:1068)
                                                        at android.os.AsyncTask$2.call(AsyncTask.java:305)
                                                        at java.util.concurrent.FutureTask.run(FutureTask.java:237)
                                                        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243)
                                                        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
                                                        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
                                                        at java.lang.Thread.run(Thread.java:760)

这个是三星S6 7.0上应用内更新抛出的异常信息

07-03 17:29:00.781 27148-28659/? E/AndroidRuntime: FATAL EXCEPTION: AsyncTask #3
                                                   Process: com.android.packageinstaller, PID: 27148
                                                   java.lang.RuntimeException: An error occurred while executing doInBackground()
                                                       at android.os.AsyncTask$3.done(AsyncTask.java:318)
                                                       at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
                                                       at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
                                                       at java.util.concurrent.FutureTask.run(FutureTask.java:242)
                                                       at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243)
                                                       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
                                                       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
                                                       at java.lang.Thread.run(Thread.java:762)
                                                    Caused by: java.lang.SecurityException: Permission Denial: opening provider com.yanzhenjie.permission.FileProvider from ProcessRecord{c869381 27148:com.android.packageinstaller/u0a122} (pid=27148, uid=10122) that is not exported from uid 11097
                                                       at android.os.Parcel.readException(Parcel.java:1693)
                                                       at android.os.Parcel.readException(Parcel.java:1646)
                                                       at android.app.ActivityManagerProxy.getContentProvider(ActivityManagerNative.java:4878)
                                                       at android.app.ActivityThread.acquireProvider(ActivityThread.java:5949)
                                                       at android.app.ContextImpl$ApplicationContentResolver.acquireUnstableProvider(ContextImpl.java:2470)
                                                       at android.content.ContentResolver.acquireUnstableProvider(ContentResolver.java:1521)
                                                       at android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:1135)
                                                       at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:988)
                                                       at android.content.ContentResolver.openInputStream(ContentResolver.java:708)
                                                       at com.android.packageinstaller.PackageInstallerActivity$StagingAsyncTask.doInBackground(PackageInstallerActivity.java:1014)
                                                       at com.android.packageinstaller.PackageInstallerActivity$StagingAsyncTask.doInBackground(PackageInstallerActivity.java:1005)
                                                       at android.os.AsyncTask$2.call(AsyncTask.java:304)
                                                       at java.util.concurrent.FutureTask.run(FutureTask.java:237)
                                                       at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243) 
                                                       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133) 
                                                       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607) 
                                                       at java.lang.Thread.run(Thread.java:762) 

--- a/permission/src/main/java/com/yanzhenjie/permission/source/ContextSource.java
+++ b/permission/src/main/java/com/yanzhenjie/permission/source/ContextSource.java
@@ -45,7 +45,7 @@ public class ContextSource extends Source {
         if (mContext instanceof Activity) {
             mContext.startActivity(intent);
         } else {
-            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
             mContext.startActivity(intent);
         }
     }
@@ -56,7 +56,7 @@ public class ContextSource extends Source {
             Activity activity = (Activity) mContext;
             activity.startActivityForResult(intent, requestCode);
         } else {
-            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
             mContext.startActivity(intent);
         }
     }

